### PR TITLE
Alternative URLs for Regions

### LINKED ISSUE #565: Alternative URLs for Regions
When entering `/cologne` it should redirect to `/koeln`

### PR DESCRIPTION
**Pull request by gill0r**
Migrate to RegionSlug, remove Region.slug
Make tests green, fix the factories
Added db seeds
Add RegionSlugs to active admin

Closes #565

### CODE REVIEW

**Comment by gill0r** [comment]
Noticed some issues while manually testing. Will commit these fixes soon.

File: app/models/region.rb [hunk 1]
```diff
@@ -1,13 +1,17 @@
...
 
   scope :active, -> { where(active: true) }
 
   def to_param
-    "#{self.id}-#{self.slug}"
+    "#{self.id}-#{self.main_slug}"
+  end
+
+  def main_slug
+    region_slugs.try(:first).try(:slug)
```
Comment (moonglum) [inline app/models/region.rb]: I'm not entirely sure if the selection of the main slug is stable. This determines on the order in which the slugs are returned by the database, right? How could I choose which slug is the main one?

**Comment by moonglum** [comment]
Thank you for your PR, this looks really solid :smile: See comment above :smile:

File: app/models/region.rb (hunk 1 above)
Comment (gill0r) [inline app/models/region.rb]: Yes, this is indeed the lazy and unstable way of implementing it ;)

I'd add a boolean `main_slug` to `RegionSlug` and make sure it's unique for the `region`. This way the main slug can be chosen manually. 
If none is chosen, the first one returned by the database could become the main_slug as a fallback.

What do you think?

Comment (moonglum) [inline app/models/region.rb]: That would be awesome :smile:

Comment (gill0r) [inline app/models/region.rb]: Done.

yisss

**Comment by bascht** [comment]
Just scrolled by & this looks really cool. Thanks @gill0r! :cake:

**Comment by moonglum** [comment]
Solid Work, @gill0r :smile: Thank you so much! :heart: Looking forward to more contributions :smile_cat:

**Pull request merged**